daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -642,94 +643,171 @@ private[lf] object Speedy {
       )
     }
 
+    @tailrec
+    private[this] def destructApp(typ: Type, tyArgs: List[Type] = List.empty): (Type, List[Type]) =
```

Comment:
didn’t we merge the same when you removed `TList` and so on? Can we share that across the value translator and here?

Reply:
:+1: